pattern/constraint/thing/RelationConstraint.java

```diff
@@ -53,6 +53,10 @@ public RelationConstraint(ThingVariable owner, List<RolePlayer> rolePlayers) {
         assert rolePlayers != null && !rolePlayers.isEmpty();
         this.rolePlayers = new ArrayList<>(rolePlayers);
         this.hash = Objects.hash(RelationConstraint.class, this.owner, this.rolePlayers);
+        for (RelationConstraint.RolePlayer rp : rolePlayers) {
+            rp.player().constraining(this);
+            rp.roleType().ifPresent(roleType -> roleType.constraining(this));
```

Comment:
out of curiosity, is there a functional difference between doing `roleType.constraining(rp)` instead of `this`? Is a `RelationConstraint.RolePlayer` a constraint itself?

Reply:
I hadn't been considering `Roleplayer` as a separate constraint, I don't thiiink that makes sense. What we need is to traverse the variables of the relation constraint, so if we used the RolePlayer that operation would become quite clumsy.